src/type.js

```diff
@@ -484,7 +484,7 @@ Sk.builtin.type.makeIntoTypeObj = function (name, t) {
     Sk.asserts.assert(name !== undefined);
     Sk.asserts.assert(t !== undefined);
     t.ob$type = Sk.builtin.type;
-    t.tp$name = name;
+    t.tp$name = "type";
```

Comment:
This is going to conflict on rebase with #1131, but that's going to be so much nicer than this :)